lib/repl.js

```diff
@@ -1080,7 +1079,11 @@ function complete(line, callback) {
     // All this is only profitable if the nested REPL does not have a
     // bufferedCommand.
     if (!magic[kBufferedCommandSymbol]) {
-      magic._domain.on('error', (err) => { throw err; });
+      magic._domain.on('error', (err) => {
+        setImmediate(() => {
+          throw err;
```

Comment:
This timeout was required for the async bootstrap to ensure the error stack was thrown correctly on an async bootstrap. We could possibly only have this immediate apply in the async case.